assert/assertions.go

```diff
@@ -1382,7 +1383,27 @@ func FileExists(t TestingT, path string, msgAndArgs ...interface{}) bool {
 	return true
 }
 
-// DirExists checks whether a directory exists in the given path. It also fails if the path is a file rather a directory or there is an error checking whether it exists.
+// NoFileExists checks whether a file does not exist in a given path. It fails
+// if the path points to an existing _file_ only.
+func NoFileExists(t TestingT, path string, msgAndArgs ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+	info, err := os.Lstat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return true
+		}
+		return true
+	}
+	if info.IsDir() {
+		return true
+	}
```

Comment:
`FileMode` gives us a little more granularity if we decide that we want this to be more subtle. Right now a symlink would also pass the check. Maybe that's fine, not sure.

Reply:
Good point. I copied this from current behaviour so that the same rules are followed (i.e. symlinks are fine). If we want to have consistency and pick one for both File and Dir, we might have compatibility-breaking changes.